### PR TITLE
Off-chain indexing is independent from workers

### PR DESCRIPTION
**Pull request by tomusdrw**
Currently the `--enable-offchain-indexing` CLI option would be taken into consideration iff off-chain workers are enabled as well.

This is incorrect, these two features serve different purposes and are designed to work separately, so it's not necessary to run off-chain workers for anything at all.

### CODE REVIEW

File: client/db/src/lib.rs [hunk 1]
```diff
@@ -34,62 +34,66 @@ pub mod offchain;
 #[cfg(any(feature = "with-kvdb-rocksdb", test))]
 pub mod bench;
 
-mod children;
 mod cache;
 mod changes_tries_storage;
+mod children;
```
Comment (bkchr) [inline client/db/src/lib.rs:39]: Rustfmt? 😜

**Review by bkchr — APPROVED** [review]

File: client/db/src/lib.rs (hunk 1 above)
Comment (tomusdrw) [reply client/db/src/lib.rs]: 🙈 😰

File: client/db/src/lib.rs [hunk 2]
```diff
@@ -699,14 +699,20 @@ pub struct BlockImportOperation<Block: BlockT> {
...
+			count += 1;
 			let key = crate::offchain::concatenate_prefix_and_key(&prefix, &key);
 			match value_operation {
 				OffchainOverlayedChange::SetValue(val) =>
 					transaction.set_from_vec(columns::OFFCHAIN, &key, val),
 				OffchainOverlayedChange::Remove =>
 					transaction.remove(columns::OFFCHAIN, &key),
 			}
+
+			if count > 0 {
```
Comment (bkchr) [inline client/db/src/lib.rs:713]: This should be moved outside of the for loop

**Comment by bkchr** [comment]
bot merge

**Pull request merged**